### DISCUSSION TITLE
How exactly heavy are x-data components for the browser?

### DISCUSSION THREAD

**Discussion by slavatarsyurimi** [alpinejs/alpine#1961] [4. General]
Hey folks, a little hypothetical question here.
I'm building a bit of a funky interactive select dropdown where each select item is a separate Alpine component. Those funky select components on the other side sit inside a datatable with a select dropdown per each row.

So for my project, in theory, I could easily end up with something like 20rows x 10 select options = 200+ Alpine x-data components on a single page (nothing too special within those components themselves).

Are browsers going to handle that fine or should I really rethink my strategy? 

Cheers.

**Comment by KevinBatdorf** [comment]
It depends. You might be fine unless all the components are updating at the same time and doing a lot of work. I would try it and see.
